checks/ulimits.go

```diff
@@ -0,0 +1,121 @@
+package checks
+
+import (
+	"fmt"
+	"syscall"
+)
+
+// UlimitChecker checks if current process resource limits are above a given minimum
+//
+// Type:
+//   - ulimit
+//
+// Supported Platforms:
+//   - MacOS
+//   - Linux
+//
+// Arguments:
+//   - item (required): A string value representing the type of limit to check
+//   - limit (required): Numerical value representing the minimum value to be tested
+//   - type (optional): "hard" or "soft" with a default of "hard"
+//
+// Notes:
+//   - Windows is not supported
+//   - "item" strings are from http://www.linux-pam.org/Linux-PAM-html/sag-pam_limits.html
+//     - Not all are supported. See `limitsByName` map below for full list
```

Comment:
The limitsByName map will not show up in the documentation. You should "duplicate" the map here in an english friendly way

Reply:
Done